optimism/src/keccak/tests.rs

```diff
@@ -199,15 +199,37 @@ fn test_keccak_fake_witness_wont_satisfy_constraints() {
     // NEGATIVIZE THE WITNESS
 
     // Break padding constraints
-    witness_env[0].witness[KeccakColumn::PadBytesFlags(0)] = Fp::from(1u32);
-    witness_env[0].constrain_padding();
+    assert_eq!(witness_env[0].is_pad(), Fp::one());
+    // Padding can only occur in suffix[3] and suffix[4] because length is 100 bytes
+    assert_eq!(witness_env[0].pad_suffix(0), Fp::zero());
+    assert_eq!(witness_env[0].pad_suffix(1), Fp::zero());
+    assert_eq!(witness_env[0].pad_suffix(2), Fp::zero());
+    // Check that the padding blocks are corrrect
+    assert_eq!(witness_env[0].block_in_padding(0), Fp::zero());
+    assert_eq!(witness_env[0].block_in_padding(1), Fp::zero());
+    assert_eq!(witness_env[0].block_in_padding(2), Fp::zero());
+    // Force claim pad in PadBytesFlags(0), involved in suffix(0)
     assert_eq!(
-        witness_env[0].errors,
-        vec![
-            Error::Constraint(PadAtEnd),
-            Error::Constraint(PaddingSuffix(0))
-        ]
+        witness_env[0].witness[KeccakColumn::PadBytesFlags(0)],
+        Fp::zero()
     );
+    witness_env[0].witness[KeccakColumn::PadBytesFlags(0)] = Fp::from(1u32);
+    // Now that PadBytesFlags(0) is 1, then block_in_padding(0) should be 0b10*
+    println!("{:?}", witness_env[0].block_in_padding(0).to_hex());
+    witness_env[0].constrain_padding();
+    // When the byte(0) is different than 0 then the padding suffix constraint also fails
+    if witness_env[0].sponge_bytes()[0] != Fp::zero() {
```

Comment:
I assume that is the main change, that in the case when it's actually /already/ zero, we expect only `PadAtEnd`.

Reply:
correct